tests/unit/test.program.js

```diff
@@ -430,6 +435,205 @@ describe('program.main', () => {
+            (message) => message.match(/bad path/)
+        ));
+        sinon.assert.notCalled(fakeApplyConfigToArgv);
+        sinon.assert.notCalled(fakeProcess.exit);
+      }));
+  });
+
+  it('throws when config file loading raises an unexpected error', () => {
+    const fakeProcess = createFakeProcess();
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      throw new Error('some error');
+    });
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    consoleStream.stopCapturing();
+    consoleStream.flushCapturedLogs();
+    consoleStream.startCapturing();
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch((error) => {
+        const {capturedMessages} = consoleStream;
+        consoleStream.stopCapturing();
+
+        if (error instanceof UsageError) {
+          throw error;
+        }
+
+        assert.match(error.message, /some error/);
+        assert.ok(capturedMessages.some(
+          (message) => message.match(/some error/)
+        ));
+        sinon.assert.notCalled(fakeApplyConfigToArgv);
+        sinon.assert.notCalled(fakeProcess.exit);
+      });
+  });
+
+  it('throws a UsageError when the loaded config can\'t be applied', () => {
+    const fakeProcess = createFakeProcess();
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {
+      throw new UsageError('bad config option');
+    });
+
+    consoleStream.stopCapturing();
+    consoleStream.flushCapturedLogs();
+    consoleStream.startCapturing();
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(UsageError, (error) => {
+        const {capturedMessages} = consoleStream;
+        consoleStream.stopCapturing();
+
+        assert.match(error.message, /bad config option/);
+        assert.ok(
+          capturedMessages.some(
+            (message) => message.match(/bad config option/)
+        ));
+        sinon.assert.notCalled(fakeProcess.exit);
+      }));
+  });
+
+  it('throws when fakeApplyConfigToArgv throws an unexpected error', () => {
```

Comment:
This can be deleted. I don't think it's helpful to check that an error is thrown.